src/main/java/frc/robot/subsystems/ShooterHoodSubsystem.java

```diff
@@ -69,12 +71,14 @@ public ShooterHoodSubsystem() {
   public void setHoodPosition(double posRequest) {
     // Used for autonomous and vision-tied control of the shooter hood.
     hoodPID.setReference(posRequest, ControlType.kPosition, 0);
+    currSetpoint = posRequest;
   }
 
   public void deployHood() {
-    if (reliableZero)
+    if (reliableZero) {
       hoodPID.setReference(MoPrefs.getShooterHoodSetpoint(), ControlType.kPosition, 0);
-    else
+      currSetpoint = MoPrefs.getShooterHoodSetpoint();
```

Comment:
I'd probably swap the order of these two lines and avoid calling MoPrefs twice.  Just call once and reuse currSetpoint in the setReference call.